### PR TITLE
Raidboss: ARF initial dungeon package

### PR DESCRIPTION
**Pull request by JLGarber**
This one's not quite as good as I would like, but I'm not sure what the best way to resolve that is. No issues with Oopsy or triggers, but unfortunately the Ascian Twins timeline is messy. Because it's a two-enemy boss, and the bosses are on separate + unequal timer loops, each phase will desync the longer it goes. Apologies for the wall of text (this is from Lahabrea, HP < 75%, but it's representative of both bosses' HP < 75% phases:)
```
2354.113: Matched entry: 2354.0 Dark Fire II (-0.113s)
2359.195: Matched entry: 2359.1 Dark Fire II (-0.095s)
2364.269: Matched entry: 2364.2 Dark Fire II (-0.069s)
2373.388: Matched entry: 2373.4 Dark Fire II (+0.012s)
2374.601: Matched entry: 2374.6 End Of Days (-0.001s)
2382.512: Matched entry: 2382.5 Shadow Flare (-0.012s)
2388.077: Matched entry: 2388.0 Dark Blizzard II (-0.077s)
2393.606: Matched entry: 2393.6 Dark Orb (-0.006s)
2398.814: Matched entry: 2398.8 Sea Of Pitch (-0.014s)
2402.372: Matched entry: 2402.4 End Of Days (+0.028s)
2406.980: Matched entry: 2407.0 Dark Orb (+0.020s)
2414.231: Matched entry: 2414.2 Dark Fire II (-0.031s)
2415.847: Matched entry: 2415.8 Dark Blizzard II (-0.047s)
2423.204: Matched entry: 2423.2 Shadow Flare (-0.004s)
2430.146: Matched entry: 2430.2 End Of Days (+0.054s)
2434.405: Matched entry: 2434.2 Dark Orb (-0.205s)
2439.410: Matched entry: 2439.4 Sea Of Pitch (-0.010s)
2443.429: Matched entry: 2443.6 Dark Blizzard II (+0.171s)
2447.751: Matched entry: 2447.5 Dark Orb (-0.251s)
2454.785: Matched entry: 2454.7 Dark Fire II (-0.085s)
2457.748: Matched entry: 2457.9 End Of Days (+0.152s)
    Jumping to 2374.600
2380.692: Matched entry: 2382.5 Shadow Flare (+1.808s)
2389.947: Matched entry: 2388.0 Dark Blizzard II (-1.947s)
2391.811: Matched entry: 2393.6 Dark Orb (+1.789s)
2398.845: Matched entry: 2398.8 Sea Of Pitch (-0.045s)
2404.236: Matched entry: 2402.4 End Of Days (-1.836s)
2405.165: Matched entry: 2407.0 Dark Orb (+1.835s)
2414.191: Matched entry: 2414.2 Dark Fire II (+0.009s)
2417.699: Matched entry: 2415.8 Dark Blizzard II (-1.899s)
2421.451: Matched entry: 2423.2 Shadow Flare (+1.749s)
2431.966: Matched entry: 2430.2 End Of Days (-1.766s)
2432.598: Matched entry: 2434.2 Dark Orb (+1.602s)
2439.452: Matched entry: 2439.4 Sea Of Pitch (-0.052s)
2445.261: Matched entry: 2443.6 Dark Blizzard II (-1.661s)
2445.889: Matched entry: 2447.5 Dark Orb (+1.611s)
2454.735: Matched entry: 2454.7 Dark Fire II (-0.035s)
2459.559: Matched entry: 2457.9 End Of Days (-1.659s)
    Jumping to 2374.600
2388.087: Matched entry: 2388.0 Dark Blizzard II (-0.087s)
    Missed sync: Shadow Flare at 2382.5 (last seen at 2378.816)
2402.426: Matched entry: 2402.4 End Of Days (-0.026s)
    Missed sync: Dark Orb at 2393.6 (last seen at 2389.931)
    Missed sync: Sea Of Pitch at 2398.8 (last seen at 2395.124)
2415.717: Matched entry: 2415.8 Dark Blizzard II (+0.083s)
    Missed sync: Dark Orb at 2407.0 (last seen at 2403.225)
    Missed sync: Dark Fire II at 2414.2 (last seen at 2410.422)
2419.543: Matched entry: 2423.2 Shadow Flare (+3.657s)
2434.337: Matched entry: 2434.2 Dark Orb (-0.137s)
    Missed sync: End Of Days at 2430.2 (last seen at 2433.7889999999998)
2439.401: Matched entry: 2439.4 Sea Of Pitch (-0.001s)
2447.522: Matched entry: 2447.5 Dark Orb (-0.022s)
    Missed sync: Dark Blizzard II at 2443.6 (last seen at 2447.068)
2454.685: Matched entry: 2454.7 Dark Fire II (+0.015s)
```

As you can see, it's mostly clean on the first loop, and then it oscillates wildly on timings, but ability order is the same. For now, I've put aggressive sync windows inside each block, so it will sort of tell the user what will happen, but it's really choppy and not-fun. Is there a better way to handle this situation, or is there an easy solution that's evading me at the moment?

Other than that, this package is ready. There was a little more manual translation work than I was expecting because this instance isn't on FFLogs, so I had to pull down the translations from XIVAPI instead. That wasn't exceedingly difficult to script, but if there are tools for that I overlooked in `translate_fight` or `check_translation`, I would much prefer that over what I did, since it was more handwork than I would like. The popup text is obviously incomplete, I just tossed in what I could find from a lazy look through the repository.

### CODE REVIEW

File: ui/raidboss/data/03-hw/dungeon/aetherochemical_research_facility.js [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+      regexFr: / 14:10EB:Horamakhet starts using Pétrification/,
+      regexJa: / 14:10EB:ハルマキス starts using ペトリファクション/,
+      infoText: {
+        en: 'Look away',
+        de: 'Wegschauen!',
+        fr: 'Ne regardez pas',
+      },
+    },
+    {
+      id: 'Facility Inertia Stream',
```
[the remark this answers is not in the document]
Comment (JLGarber) [reply ui/raidboss/data/03-hw/dungeon/aetherochemical_research_facility.js]: That's where Harmachis chains its target for Ballistic Missile. After about 4 seconds, Ballistic Missile goes off, dealing split damage to everyone in the stack. The moment Ballistic Missile goes off, a meteor laser head marker is placed on the target, and 4 seconds later they are struck for moderate damage. It's a chain of mechanics that requires reactive healing if it's not handled correctly, but can be basically ignored if someone else stacks in.

File: ui/raidboss/data/03-hw/dungeon/aetherochemical_research_facility.txt [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+444.0 "Magitek Slug"
+444.7 "Aetherochemical Grenado"
+449.9 "Aetherochemical Grenado"
+450.4 "Magitek Slug"
+454.1 "Magitek Slug"
+455.1 "Aetherochemical Grenado"
+457.8 "Magitek Slug"
+
+#~~~~~~~~~~~#
+# Harmachis #
```
[the remark this answers is not in the document]
Comment (JLGarber) [reply ui/raidboss/data/03-hw/dungeon/aetherochemical_research_facility.txt]: Regula most definitely deserved better things than what Square gave him. (Or did you mean Harmachis?)

[the remark this answers is not in the document]
Comment (quisquous) [reply ui/raidboss/data/03-hw/dungeon/aetherochemical_research_facility.txt]: Yeah, I meant Harmachis!

File: ui/raidboss/data/03-hw/dungeon/aetherochemical_research_facility.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (quisquous) [reply ui/raidboss/data/03-hw/dungeon/aetherochemical_research_facility.js]: This is the t8 ballistic missile mechanic where you sometimes need exactly 2 or exactly 3 depending on the circle size? (Or is it only the 2 person stack in ARF?)

If this is what I think it is, I think the callout is "one person stack on X" (and it'd be fine to say that to the person caught too even if they can't do anything).

?

**Comment by JLGarber** [comment]
From what I've seen, each phase ( Igeyorhm HP > 75%, Igeyorhm HP < 75%, Lahabrea HP > 75%, and Lahabrea HP < 75%) re-syncs both Ascians at its beginning.

> (1) only add a timeline for one of them and triggers if needed for the other
(2) add timelines for both the first loop and only a timeline for one the next time around
(3) -funroll-loops and just kind of write out the timeline manually for a couple of iterations (about where folks would probably ever see to) and then maybe loop just on one of them if you really want to have it loop after that

On option 1, that might be workable. For both Ascians, when in their 100%-75% phases, the only thing the inactive one casts is End of Days. During the <75% phases, the inactive one adds their level two elemental spell, which is just random circle AoEs.

That being said, option 3 is probably the "better" solution in this case, since anyone who's running Cactbot is *likely* to be capable of pushing enough damage to clear the phase before a long loop resets. It feels a bit icky, but for 95% of users it will probably be a better experience. If you're comfortable with it, I'll resubmit with longer phase loops once the servers come back up.

> There is not anything that uses xivapi. I think it would be great if it could and would wholeheartedly support such a pull request. That's probably a superior option to "translating" via fflogs, which was always a bit of a hack. The only real advantage to fflogs is that it maybe makes it easier to look up mob name translations. Abilities and effects at least have ids that you can look up in xivapi, but there's no npc id in network logs. This sort of feature would also be super great at the start of new tiers when ACT hasn't updated with new ability names but xivapi is updated.

I'll open an issue about that then. We have maybe what, 1.5 months to next patch? I might prioritize that over other utilities in that case. In the case of the ARF translations, I hacked something together to run in the browser console because I'm more familiar with Javascript's XMLHTTPRequest and it was intended as a one-off, but it can't be that difficult in Python. It would also be an improvement over FFLogs translating, since we could just translate the abilities that appear in the timeline without having to cull the results from FFLogs.

File: ui/raidboss/data/03-hw/dungeon/aetherochemical_research_facility.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (JLGarber) [reply ui/raidboss/data/03-hw/dungeon/aetherochemical_research_facility.js]: That seems likely, yes. It's a ground marker, and then the overhead marker for Gaseous Bomb is the same as the one in T8, so I think we can assume that's the case.

**Comment by JLGarber** [comment]
Alright, this should solve that issue. It's still choppy on the 100-75% HP sections, but it's within 0.5s, so I think that's acceptable. The HP < 75% sections smoothed out nicely. It all feels so icky, but it's probably the best solution for a problem that's *likely* never going to come up.

This also fixes a bug with Ascian Prime's final phase, where it was jumping to the wrong tank buster and causing some odd timeline jank.

**Comment by quisquous** [comment]
Thanks! Yeah, I understand that solution doesn't feel great, but I think it's the best that can be done with the tools at hand.

**Pull request merged**